### ISSUE TITLE
Refactor settings to use PreferenceFragments instead of PreferenceActivity

### ISSUE THREAD

**Issue by moving-bits** [cgeo/cgeo#10579]
## Help needed
While deep-diving into the theming issue (#9373) yesterday and playing around with different strategies, I've stumbled upon our outdated preferences implementation. We are still using `PreferenceActivity` with several preference screens, which is deprecated for a long time. What is worse is that this is technically not compatible to the rest of our app, which is using AndroidX and `AppCompatActivity` - which means a lot of extra work and inconsistencies.

As I have zero experience with fragment handling I'm asking for help, so if someone is willing to migrate our preferences from `PreferenceActivity` to `AppCompatActivity` and `PreferenceFragments` (or even some full up-to-date AndroidX way), please go ahead, as this will help us moving forward towards a consistent theming approach.

**Comment by fm-sys** [comment]
I've implemented a preference activity with AndroidX for two other Apps I wrote from scratch. I can try to, but I'm not sure if will have enough time to refactor this. Btw, it seams like multi screen preferences aren't really supported anymore which means we have to implement multiple activities/fragments...

**Comment by fm-sys** [comment]
@rsudev you have started refactoring our preferences some time ago, haven't you? Is there any progress?

**Comment by bekuno** [comment]
@SchoolGuy As you migrated the settings in WhereYouGo to Fragments (PR https://github.com/cgeo/WhereYouGo/pull/240), can you support or implement this also for c:geo?

**Comment by SchoolGuy** [comment]
@bekuno Certainly yes. I am a bit busy catching up with things since I wrote my final exams on Wednesday but I guess I can come up with something in the next two weeks.

**Comment by bekuno** [comment]
@SchoolGuy  I wish you that your final exams were very good. Feel free to help whenever you have some spare time :)

**Comment by moving-bits** [comment]
Setting prio to "high" here, as using the old `PreferenceActivity` is not compatible to material design, not even to `AppCompatActivity`, and therefore requires lots of extra code and ugly workarounds to work at all, and even with those workarounds does not fit completely into our new material look...

I guess we should do this in several steps: Start with a single preference screen, migrate it to `PreferenceFragment`, and have it as a parallel menu entry besides our current settings. And then migrate screen by screen until the old preferences are gone.

If someone has the time to build a basic frame for one or two settings screens, we can continue from there in parallel to migrate the others as well... (Sadly I haven't found the time yet to look into what @rsudev has already done while working on the Sailfish issue, but anyway, fragments are pretty unknown to me.)

**Comment by SchoolGuy** [comment]
@moving-bits My WhereYouGo PR is merged soon (only review corrections to do tomorrow). Then I will get started on this immediately.

**Comment by moving-bits** [comment]
> @moving-bits My WhereYouGo PR is merged soon (only review corrections to do tomorrow). Then I will get started on this immediately.

Big thumbs up, would be great! 🚀

**Comment by TacoTheDank** [comment]
I would also be willing to try my hand at this, just as a backup resource in case something changes with @SchoolGuy's schedule or something like that :)

**Comment by SchoolGuy** [comment]
Starting tomorrow. But if you want we can collaborate to make it quicker. @TacoTheDank

**Comment by TacoTheDank** [comment]
@SchoolGuy Sure! I am available to provide code review, suggestions, improvements, and things like that on GitHub as the PR is made. Wishing you best of luck in this process, cause this refactor seems like quite a doozy lol

**Comment by SchoolGuy** [comment]
Soooo I thought the WhereYouGo settings were complicated from a code perspective. I was wrong. This is not a "just-do-it-again" sadly. I will make myself familiar with the codebase and try to refactor. Especially all the static stuff will make things complicated...

**Comment by SchoolGuy** [comment]
P.S.: Once the first step is done I will open a WIP PR so you can watch my progress
